tests/framework/e2e/etcd_process.go

```diff
@@ -230,7 +233,7 @@ func (ep *EtcdServerProcess) Config() *EtcdServerProcessConfig { return ep.cfg }
 
 func (ep *EtcdServerProcess) Logs() LogsExpect {
```

Comment:
Code looks nice, but changing `func (ep *EtcdServerProcess) Logs() LogsExpect` semantic behavior is risky. When I expect most people will assume `member.Logs().ExpectWithContext()` will analyse logs emitted from the current run and might stumble on problem that log they are expecting was already emitted before restart. 

Proposal:
* Have `func (ep *EtcdServerProcess) Logs() LogsExpect` return logsExpect from the latest run `ep.logs.procLogs[len(ep.logs.procLogs)-1]`
* Have a separate method like `allLogLines` or `aggregatedLogLines` etcd that will return appended results of all runs.

What do you think?

Reply:
Agreed with the incremental changes to be safe. 

`func (ep *EtcdServerProcess) Logs() LogsExpect` was using `ep.proc` in happy case and most of the cases. 

The PR is intended to retain all the previous runs logs after all the runs of the process are stopped. 

Seems only `func (m *multiLogsExpect) Lines()` is needed.

Reply:
> Have func (ep *EtcdServerProcess) Logs() LogsExpect return logsExpect from the latest run ep.logs.procLogs[len(ep.logs.procLogs)-1]

Sounds like a good proposal to me. Is that also acceptable for you @chaochn47?

> Seems only func (m *multiLogsExpect) Lines() is needed.

Currently yes, I'm surprised we don't have expect tests across stop/restart scenarios.

Reply:
> Currently yes, I'm surprised we don't have expect tests across stop/restart scenarios.

We want to explicitly avoid that because it not natural. Imagine that you want to check that etcd member has started using one of the starting logs. It's natural to call `member.Logs().Expect("etcd started")`, but it needs to only consider the logs from the last restart to make sense. Otherwise it might always return true when member was already restarted.

tldr. Don't want to change the semantics of `func (ep *EtcdServerProcess) Logs() LogsExpect`.

Reply:
Cool, pushed another by just moving up the Lines and Count via interface, so the current LogsExpect stays entirely untouched.